catch-uncommitted.sh

```diff
@@ -1,9 +1,24 @@
 #!/bin/sh
 set -e
 
+catch_no_git=0
+
+while [ $# -gt 0 ]; do
+	key=$1
+	shift
+	case $key in
+		--catch-no-git)
+			catch_no_git=1
+		;;
+	esac
+done
+
 echo
 
-if ! git diff HEAD --exit-code ; then
+if [ $catch_no_git -eq 1 ] && !(hash git > /dev/null); then
```

Comment:
My shell skills don't comprehend `hash git`, what is it supposed to output if not directed to dev/null?

Reply:
When the command is not found it returns `bash: hash: myNotFoundCommand: not found` otherwise it doesn't print anything. It's the exit status though that's useful here.
I actually have to change `>` to `2>` though, since it seems that `>` doesn't pipe the stderror but only the stdout.
See: https://stackoverflow.com/questions/592620/how-to-check-if-a-program-exists-from-a-bash-script

Reply:
That is what I assumed, just wanted to check. I was going to ask "Why don't you use `which`", but the link you posted explained that to me as well 😄